cidc_api/models/models.py

```diff
@@ -945,6 +945,38 @@ def get_summaries(session: Session) -> List[dict]:
             group by trial_id
         """
 
+        # Compute the number of samples associated with elisa uploads.
+        # Unlike other assays, elisa metadata is an array of entries, each containing a single data file.
+        # The number of samples corresponding to a given entry is stored like:
+        # entry["assay_xlsx"]["number_of_samples"].
+        elisa_subquery = """
+            select
+                trial_id,
+                'elisa' as key,
+                sum((entry#>'{assay_xlsx,number_of_samples}')::text::integer) as value
+            from
+                trial_metadata,
+                jsonb_array_elements(metadata_json#>'{assays,elisa}') entry
+            group by trial_id
+        """
+
+        # Compute the number of samples associated with nanostring uploads.
+        # Nanostring metadata has a slightly different structure than typical
+        # assays, where each batch has an array of runs, and each run has
+        # an array of sample-level entries.
+        cytof_subquery = """
+            select
+                trial_id,
+                'cytof' as key,
+                sum(jsonb_array_length(batches->'records')) as value
+            from
+                trial_metadata,
+                jsonb_each(metadata_json->'assays') assays,
+                jsonb_array_elements(value) batches
+            where key in ('cytof_10021', 'cytof_e4412')
+            group by trial_id
+        """
```

Comment:
Since we're just renaming the `cytof_10021` and `cytof_e4412` assay types to `cytof` here, I'm not convinced this needs a separate subquery - I think we should only create separate subqueries for assays that deviate from the typical "batches of sample-level records" shape. 

Instead, I'd prefer if we introduce this change (and other assay renamings, like `'hande' -> 'h&e'`) in the `generic_assay_subquery` like:
```python
generic_assay_subquery = """
    select
        trial_id,
        case
            when key in ('cytof_e4412', 'cytof_10021') then 'cytof'
            when key = 'hande' then 'h&e'
            else key
        end as key,
        sum(jsonb_array_length(batches->'records')) as value
    from
        trial_metadata,
        jsonb_each(metadata_json->'assays') assays,
        jsonb_array_elements(value) batches
    where key not in ('olink', 'nanostring')
    group by trial_id, key
"""
```

Reply:
oops, I forgot to add the different handling for e4412 because there's a participants level to go down too

Reply:
Ooo, hang on. I don't think the `group by key` will work in the example code I wrote above (pretty sure the different `cytof_*`s will be grouped separately). I do still wonder if there's a no-extra-subquery way to do it, though.

Reply:
I can do this for hande and cytof_10021, then just have a subquery for cytof_e4412 because of the different structure?

Reply:
[Resolved offline - subsequent solution in the PR represents what we discussed]